include/abstract_index.h

```diff
@@ -79,6 +79,9 @@ class AbstractIndex
                                                       const size_t K, const uint32_t L, IndexType *indices,
                                                       float *distances);
 
+    template <typename data_type, typename tag_type, typename label_type>
+    int insert_point(const data_type *point, const tag_type tag, const std::vector<label_type> &labels = {});
+
```

Comment:
should the base class AbstractIndex have insert with labels? What if some one wants to write an index that does not support labels?

Reply:
There should be a method to do just that, its right below.